### ISSUE TITLE
Error with gulp while npm install

### ISSUE THREAD

**Issue by derweili** [olefredrik/FoundationPress#1322]
Recently I get an error while running `npm install`.
Is error ist related to gulp.

**How can this bug be reproduced?**

1. clone FoundationPress
2. run `npm install`

**What happened**
While installing all npm dependencies I get following error:

```
WerbeagtensiMac:FoundationPress werbeagenten$ npm install
Unhandled rejection Error: Command failed: /usr/bin/git checkout 4.0ract fancy-log@1.3.2
error: pathspec '4.0' did not match any file(s) known to git.

    at ChildProcess.exithandler (child_process.js:198:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:920:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:230:5)
```

**My Solution**
I was able to solve this error by removing the `#4.0` at the gulp devDependencies.

https://github.com/olefredrik/FoundationPress/blob/b0bc01e3bfc8640771a47f9cc9cbe632205d43d7/package.json#L42

**Please List the Following:**
* OS & version: macOS 10.13.6
* Node version (node -v) [Node v4+ is required] : v6.11.5 (installed via nvm)
* FoundationPress version: 2.10.4
* Foundation version : 6.4.3

**Comment by melek** [comment]
I also solved this issue manually, though I did it by replacing the line with:
`"gulp": "^4.0", `

I'm not proficient enough with npm to understand why both worked and which solution is the 'correct' one, but wanted to add my experience in case it informs a better bugfix.

**Comment by dantahoua** [comment]
Same here... Depending of the npm version (I use nvm), error is sightly different but always around this gulp 4.0... With node 6.11.5 I have:
`npm ERR! git rev-list -n1 4.0: fatal: ambiguous argument '4.0': unknown revision or path not in the working tree.
npm ERR! git rev-list -n1 4.0: Use '--' to separate paths from revisions, like this:
npm ERR! git rev-list -n1 4.0: 'git <command> [<revision>...] -- [<file>...]'
npm ERR! git rev-list -n1 4.0: 
npm ERR! git rev-list -n1 4.0: fatal: ambiguous argument '4.0': unknown revision or path not in the working tree.
npm ERR! git rev-list -n1 4.0: Use '--' to separate paths from revisions, like this:
npm ERR! git rev-list -n1 4.0: 'git <command> [<revision>...] -- [<file>...]'
npm ERR! git rev-list -n1 4.0: 
The authenticity of host 'github.com (192.30.253.113)' can't be established.
RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
Are you sure you want to continue connecting (yes/no)? yes
...
..
.
npm ERR! node v6.11.5
npm ERR! npm  v3.10.10
npm ERR! code 128

npm ERR! Command failed: git clone --template=/Users/studio12/.npm/_git-remotes/_templates --mirror git@github.com:gulpjs/gulp.git /Users/studio12/.npm/_git-remotes/git-github-com-gulpjs-gulp-git-4-0-7c06e801
npm ERR! Cloning into bare repository '/Users/studio12/.npm/_git-remotes/git-github-com-gulpjs-gulp-git-4-0-7c06e801'...
npm ERR! Warning: Permanently added 'github.com,192.30.253.113' (RSA) to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.` 

Seems the repository is not here, or an access problem/authorization...

**Comment by dantahoua** [comment]
I found it after reading this post on the Gulp GIT : [https://github.com/gulpjs/gulp/issues/1486](url)
Yo have to replace the gulp line in `package.json` with
`"gulp": "next",`

**Comment by derweili** [comment]
> I found it after reading this post on the Gulp GIT : [https://github.com/gulpjs/gulp/issues/1486](url)
> Yo have to replace the gulp line in `package.json` with
> `"gulp": "next",`

I don't think "next" is the right version to reference. "next" does not reference to a unique version but to the latest stable version. This version will change in the future. 
Using "next" could cause incompatibility problems when gulp has breaking changes in upcoming versions.

That's why i prefer @melek's  solutions:
> I also solved this issue manually, though I did it by replacing the line with:
> `"gulp": "^4.0", `

**Comment by neilwave** [comment]
I tested an older installation i had (same Foundation Press Version, 2.10.4), just copied all files into the new folder were i had the `Unhandled rejection Error` before and npm start, npm build are running well.
package.json: line 42: `"gulp": "gulpjs/gulp#4.0",` is unchanged.

But that can´t be the solution.

Next try:
I overlocked that my current node version was higher than recommended, so i downgraded it  to 6.11.5.
Changed package.json "gulp": "^4.0" (like @melek)

now it´s running
